nhais-adaptor/adaptor/outgoing/fhir_helpers/operation_definition.py

```diff
@@ -0,0 +1,166 @@
+from fhirclient.models.humanname import HumanName
+from fhirclient.models.identifier import Identifier
+from fhirclient.models.operationdefinition import OperationDefinitionParameterBinding, OperationDefinitionParameter, \
+    OperationDefinition
+from fhirclient.models.patient import Patient
+from fhirclient.models.practitioner import Practitioner
+from fhirclient.models.address import Address
+from fhirclient.models.extension import Extension
+
+
+class OperationDefinitionHelper:
+
+    @staticmethod
+    def create_parameter_with_binding(name, value):
+        """
+        Helper function to create a fhir parameter with binding value
+        :param name: name of the parameter
+        :param value: the value of the parameter
+        :return: OperationDefinitionParameter
+        """
+        identifier = Identifier({'value': value})
+        op_param_binding = OperationDefinitionParameterBinding(
+            {'strength': 'required', 'valueSetReference': {'identifier': identifier.as_json()}}
+        )
+        op_param = OperationDefinitionParameter(
+            {'name': name, 'use': 'in', 'min': 1, 'max': '1',
+             'binding': op_param_binding.as_json()})
+
+        return op_param
+
+    @staticmethod
+    def create_parameter_with_resource_ref(name, resource_type, reference):
+        """
+        Helper function to create a parameter with a reference to a resource
+        :param name: the name of the parameter
+        :param resource_type: the type of resource
+        :param reference: reference to the resource
+        :return: OperationDefinitionParameter
+        """
+        op_param = OperationDefinitionParameter(
+            {'name': name, 'use': 'in', 'min': 1, 'max': '1', 'type': resource_type,
+             'profile': {'reference': f"#{reference}"}})
+        return op_param
+
+    @staticmethod
+    def create_operation_definition(name, code, date_time, contained, parameter):
+        """
+        Helper function to create a fhir operation definition
+        :param name: The name of the operation
+        :param code: The code of the operation
+        :param date_time: the time stamp of the operation
+        :param contained: list of contained resources
+        :param parameter: list of parameters

```

Comment:
I wanted to keep it close to the Fhir models as possible to avoid confusion. Since it is called `parameter` in the model even though it is a list. However, I suppose this helper function abstracts some of that fhir craziness so will change to `parameters`